src/Handling/SpellController.php

```diff
@@ -169,13 +160,16 @@ public function index()
         }
 
         // Check params and request type
-        if (!Director::is_ajax() || empty($data['method']) || empty($data['params']) || count($data['params']) < 2) {
+        if (!Director::is_ajax() || empty($data['method']) || empty($data['lang'])) {
             return $this->error(_t(__CLASS__ . '.InvalidRequest', 'Invalid request'), 400);
         }
 
         // Check locale
-        $params = $data['params'];
-        $locale = $params[0];
+        $locale = $data['lang'];
+
+        // Check if the locale is actually a language
+        $locale = i18n::getData()->localeFromLang($locale);
```

Comment:
Does the 'lang' ever come as the full locale form? e.g. `en_NZ`. localeFromLang() doesn't work well if you pass in the full locale already.

Reply:
Yes, it can come as a locale. I'd assumed this method would handle that. Do you have a suggestion other than what I had originally?

Reply:
localeFromLang implies you are sending a lang (e.g. `en`) as an argument, and doesn't validate that. You should probably check that manually here.